r3bbase/R3BFileSource.cxx

```diff
@@ -357,6 +375,45 @@ Bool_t R3BFileSource::Init()
         fInputFile.seekg(0, std::ios::beg);
     }
 
+    if (fInputFile.is_open())
+    {
+        LOG(INFO) << "R3BFileSource::Init() Reading RunId file";
+        std::string buffer;
+        char* p;
+        do
+        {
+            getline(fInputFile, buffer);
+            LOG(INFO) << "read from file: \"" << buffer << "\"";
+            if (buffer.find("fRunID") == 0)
+            {
+                buffer.erase(0, buffer.find('=') + 1);
+
+                frunid[array_size] = strtol(buffer.c_str(), &p, 16);
+                LOG(DEBUG) << "RuniD= " << frunid[array_size];
+            }
+            if (buffer.find("FILE") == 0)
+                continue;
+
+            getline(fInputFile, buffer);
+            LOG(INFO) << "read from file: \"" << buffer << "\"";
+
+            std::string buf1 = buffer.c_str();
+            buf1.erase(buf1.find(' '), buf1.length());
+            buffer.erase(0, buffer.find(' ') + 1);
+            fexpid = strtol(buf1.c_str(), &p, 16);
```

Comment:
You should check, that after a call to `strtol` the value returned in `p` is different from the value passed in as `buf1.c_str()`. If that is not the case (i.e. if `p == buf1.c_str()` is true), then no value was converted.

Reply:
This also applies to the other invocations of `strtol`